core/trino-main/src/main/java/io/trino/execution/buffer/PageDeserializer.java

```diff
@@ -238,6 +238,76 @@ public void readBytes(byte[] destination, int destinationIndex, int length)
             }
         }
 
+        @Override
+        public void readShorts(short[] destination, int destinationIndex, int length)
+        {
+            ReadBuffer buffer = buffers[0];
+            int shortsRemaining = length;
+            while (shortsRemaining > 0) {
+                ensureReadable(min(Long.BYTES, shortsRemaining * Short.BYTES));
```

Comment:
Not sure I understand this line. If there are, say, 100 shorts remaining, this would do:

`ensureReadable(min(8, 200))` -> `ensureReadable(8)`

Also, what's the significance of `Long.BYTES` here? Why would this method care about having at least 8 readable bytes, assuming that's the intention?

Reply:
Absolutely agree.  This is the style of the other methods in this class I had to ask @arhimondr how this code works, and this is what he said:

> Hmm, it does look weird. I need to read it carefully to tell you more.
> It basically tells that "at least" 8 bytes (or less if remaining bytes is less than 8" should be available
> If not available - it will read an entire encryption / compression block

I think we all agree it need some renaming